mysql-connector-cpp fails to build on amd64

The package mysql-connector-cpp with version 9.3.0 and epoch 0 fails to build from source on amd64 with the following error:

```
2025-06-02 08:33:24.046076927 +0000 UTC   665 |   add_option(Session_option_impl::PORT, val);
2025-06-02 08:33:24.04608227 +0000 UTC       |   ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025-06-02 08:33:24.04608729 +0000 UTC /home/build/include/mysqlx/common/value.h:168:21: error: expected primary-expression
2025-06-02 08:33:24.046091833 +0000 UTC   168 |     : Value(uint64_t(val))
2025-06-02 08:33:24.046097418 +0000 UTC       |             ~~~~~~~~^~~~~
2025-06-02 08:33:24.046102792 +0000 UTC /home/build/include/mysqlx/common/value.h: In instantiation of 'mysqlx::abi2::r0::common::Value::Value(T) [with T = long unsigned int; typename std::enable_if<std::is_unsigned<_Tp>::value>::type* <anonymous> = 0]':
2025-06-02 08:33:24.046108429 +0000 UTC /home/build/xapi/../common/settings.h:1115:18:   required from 'void mysqlx::abi2::r0::common::Settings_impl::Setter::add_option(int, const T&) [with T = long unsigned int]'
2025-06-02 08:33:24.046114322 +0000 UTC  1115 |       it->second = val;
2025-06-02 08:33:24.046119576 +0000 UTC       |       ~~~~~~~~~~~^~~~~
2025-06-02 08:33:24.04612467 +0000 UTC /home/build/xapi/../common/settings.h:796:13:   required from here
2025-06-02 08:33:24.04613001 +0000 UTC   796 |   add_option(Settings_impl::Session_option_impl::CONNECT_TIMEOUT, timeout);
2025-06-02 08:33:24.046135772 +0000 UTC       |   ~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2025-06-02 08:33:24.04614204 +0000 UTC /home/build/include/mysqlx/common/value.h:168:21: error: expected primary-expression
2025-06-02 08:33:24.046148238 +0000 UTC   168 |     : Value(uint64_t(val))
2025-06-02 08:33:24.046154718 +0000 UTC       |             ~~~~~~~~^~~~~
2025-06-02 08:33:24.04615992 +0000 UTC /home/build/include/mysqlx/common/value.h: In instantiation of 'mysqlx::abi2::r0::common::Value::Value(T) [with T = short unsigned int; typename std::enable_if<std::is_unsigned<_Tp>::value>::type* <anonymous> = 0]':
2025-06-02 08:33:24.046165408 +0000 UTC /home/build/xapi/../common/settings.h:1115:18:   required from 'void mysqlx::abi2::r0::common::Settings_impl::Setter::add_option(int, const T&) [with T = short unsigned int]'
2025-06-02 08:33:24.046170794 +0000 UTC  1115 |       it->second = val;
2025-06-02 08:33:24.04617811 +0000 UTC       |       ~~~~~~~~~~~^~~~~
2025-06-02 08:33:24.046184083 +0000 UTC /home/build/xapi/../common/settings.h:264:15:   required from 'void mysqlx::abi2::r0::common::Settings_impl::Setter::set_option(const T&) [with int OPT = 3; T = short unsigned int]'
2025-06-02 08:33:24.046190282 +0000 UTC   264 |     add_option(OPT, val);
2025-06-02 08:33:24.046197025 +0000 UTC       |     ~~~~~~~~~~^~~~~~~~~~
2025-06-02 08:33:24.046202625 +0000 UTC /home/build/xapi/../common/settings.h:1327:40:   required from here
2025-06-02 08:33:24.046209604 +0000 UTC  1327 |   set_option<Session_option_impl::PORT>(port);
2025-06-02 08:33:24.046215805 +0000 UTC       |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~
2025-06-02 08:33:24.046221301 +0000 UTC /home/build/include/mysqlx/common/value.h:168:21: error: expected primary-expression
2025-06-02 08:33:24.046227128 +0000 UTC   168 |     : Value(uint64_t(val))
2025-06-02 08:33:24.046233283 +0000 UTC       |             ~~~~~~~~^~~~~
2025-06-02 08:33:24.994097311 +0000 UTC In file included from /home/build/include/mysqlx/devapi/../common.h:53,
2025-06-02 08:33:24.994101825 +0000 UTC ninja: job failed: /usr/local/bin/c++ -DCONCPP_BUILD_SHARED -I/home/build/include -I/home/build/output/include/mysqlx -I/home/build/cdk/include -I/home/build/output/cdk/include -I/home/build/cdk/extra/rapidjson/include -I/home/build/cdk/parser -Wall  -O3 -DNDEBUG -std=c++17 -fPIC -fvisibility=hidden -Wno-stringop-overflow -DRAPIDJSON_HAS_CXX11_NOEXCEPT=1 -DRAPIDJSON_HAS_CXX11_RANGE_FOR=1 -DRAPIDJSON_HAS_CXX11_RVALUE_REFS=1 -MD -MT devapi/CMakeFiles/devapi.dir/crud.cc.o -MF devapi/CMakeFiles/devapi.dir/crud.cc.o.d -o devapi/CMakeFiles/devapi.dir/crud.cc.o -c /home/build/devapi/crud.cc
2025-06-02 08:33:24.994269776 +0000 UTC                  from /home/build/include/mysqlx/devapi/common.h:35,
2025-06-02 08:33:24.994349581 +0000 UTC                  from /home/build/include/mysqlx/xdevapi.h:93,
2025-06-02 08:33:24.99438074 +0000 UTC                  from /home/build/devapi/crud.cc:31:
2025-06-02 08:33:24.994430104 +0000 UTC /home/build/include/mysqlx/devapi/../common/value.h:100:5: error: 'uint64_t' does not name a type
2025-06-02 08:33:24.994545654 +0000 UTC   100 |     uint64_t v_uint;
2025-06-02 08:33:24.994575534 +0000 UTC       |     ^~~~~~~~
2025-06-02 08:33:24.994594682 +0000 UTC /home/build/include/mysqlx/devapi/../common/value.h:37:1: note: 'uint64_t' is defined in header '<cstdint>'; this is probably fixable by adding '#include <cstdint>'
2025-06-02 08:33:24.994632344 +0000 UTC    36 | #include "error.h"
2025-06-02 08:33:24.99465561 +0000 UTC   +++ |+#include <cstdint>
2025-06-02 08:33:24.994675652 +0000 UTC    37 | #include "util.h"
2025-06-02 08:33:24.994750142 +0000 UTC /home/build/include/mysqlx/devapi/../common/value.h:138:17: error: expected ')' before 'v'
2025-06-02 08:33:24.994779407 +0000 UTC   138 |   Value(uint64_t v) : m_type(UINT64)
2025-06-02 08:33:24.994803074 +0000 UTC       |        ~        ^~
2025-06-02 08:33:24.994846124 +0000 UTC       |                 )
2025-06-02 08:33:24.994873872 +0000 UTC /home/build/include/mysqlx/devapi/../common/value.h:198:3: error: 'uint64_t' does not name a type
2025-06-02 08:33:24.994892964 +0000 UTC   198 |   uint64_t get_uint() const
2025-06-02 08:33:24.994932824 +0000 UTC       |   ^~~~~~~~
2025-06-02 08:33:24.994961162 +0000 UTC /home/build/include/mysqlx/devapi/../common/value.h:198:3: note: 'uint64_t' is defined in header '<cstdint>'; this is probably fixable by adding '#include <cstdint>'
2025-06-02 08:33:24.995023232 +0000 UTC /home/build/include/mysqlx/devapi/../common/value.h: In member function 'bool mysqlx::abi2::r0::common::Value::get_bool() const':
2025-06-02 08:33:24.995054186 +0000 UTC /home/build/include/mysqlx/devapi/../common/value.h:191:36: error: 'const union mysqlx::abi2::r0::common::Value::<unnamed>' has no member named 'v_uint'; did you mean 'v_sint'?
2025-06-02 08:33:24.99507366 +0000 UTC   191 |     case UINT64: return 0 != m_val.v_uint;
2025-06-02 08:33:24.995110971 +0000 UTC       |                                    ^~~~~~
2025-06-02 08:33:24.99513816 +0000 UTC       |                                    v_sint
2025-06-02 08:33:24.995161083 +0000 UTC /home/build/include/mysqlx/devapi/../common/value.h: In member function 'int64_t mysqlx::abi2::r0::common::Value::get_sint() const':
2025-06-02 08:33:24.995198222 +0000 UTC /home/build/include/mysqlx/devapi/../common/value.h:219:5: error: 'uint64_t' was not declared in this scope
2025-06-02 08:33:24.995227266 +0000 UTC   219 |     uint64_t val = get_uint();
2025-06-02 08:33:24.995245501 +0000 UTC       |     ^~~~~~~~
2025-06-02 08:33:24.995284205 +0000 UTC /home/build/include/mysqlx/devapi/../common/value.h:219:5: note: 'uint64_t' is defined in header '<cstdint>'; this is probably fixable by adding '#include <cstdint>'
2025-06-02 08:33:24.995313573 +0000 UTC /home/build/include/mysqlx/devapi/../common/value.h:221:36: error: 'val' was not declared in this scope; did you mean 'm_val'?
2025-06-02 08:33:24.99533236 +0000 UTC   221 |     if (!check_num_limits<int64_t>(val))
2025-06-02 08:33:24.995372634 +0000 UTC       |                                    ^~~
2025-06-02 08:33:24.99540033 +0000 UTC       |                                    m_val
2025-06-02 08:33:24.995418275 +0000 UTC /home/build/include/mysqlx/devapi/../common/value.h:224:12: error: 'val' was not declared in this scope; did you mean 'm_val'?
2025-06-02 08:33:24.995456899 +0000 UTC   224 |     return val;
2025-06-02 08:33:24.995483742 +0000 UTC       |            ^~~
2025-06-02 08:33:24.995506623 +0000 UTC       |            m_val
2025-06-02 08:33:24.995537368 +0000 UTC /home/build/include/mysqlx/devapi/../common/value.h: In member function 'float mysqlx::abi2::r0::common::Value::get_float() const':
2025-06-02 08:33:24.995566237 +0000 UTC /home/build/include/mysqlx/devapi/../common/value.h:232:36: error: 'const union mysqlx::abi2::r0::common::Value::<unnamed>' has no member named 'v_uint'; did you mean 'v_sint'?
2025-06-02 08:33:24.995585095 +0000 UTC   232 |     case UINT64: return 1.0F*m_val.v_uint;
2025-06-02 08:33:24.995603627 +0000 UTC       |                                    ^~~~~~
2025-06-02 08:33:24.995620636 +0000 UTC       |                                    v_sint
2025-06-02 08:33:24.995666038 +0000 UTC /home/build/include/mysqlx/devapi/../common/value.h: In member function 'double mysqlx::abi2::r0::common::Value::get_double() const':
2025-06-02 08:33:24.995689319 +0000 UTC /home/build/include/mysqlx/devapi/../common/value.h:244:35: error: 'const union mysqlx::abi2::r0::common::Value::<unnamed>' has no member named 'v_uint'; did you mean 'v_sint'?
2025-06-02 08:33:24.995709935 +0000 UTC   244 |     case UINT64: return 1.0*m_val.v_uint;
2025-06-02 08:33:24.99575842 +0000 UTC       |                                   ^~~~~~
2025-06-02 08:33:24.995777283 +0000 UTC       |                                   v_sint
2025-06-02 08:33:24.995794719 +0000 UTC In file included from /home/build/include/mysqlx/devapi/result.h:40,
2025-06-02 08:33:24.995843567 +0000 UTC                  from /home/build/include/mysqlx/xdevapi.h:94:
2025-06-02 08:33:24.995863426 +0000 UTC /home/build/include/mysqlx/devapi/document.h: In member function 'T mysqlx::abi2::r0::Value::get() const [with T = unsigned int]':
2025-06-02 08:33:24.995886149 +0000 UTC /home/build/include/mysqlx/devapi/document.h:775:20: error: 'get_uint' was not declared in this scope; did you mean 'get_sint'?
2025-06-02 08:33:24.995929941 +0000 UTC   775 |     uint64_t val = get_uint();
2025-06-02 08:33:24.995959129 +0000 UTC       |                    ^~~~~~~~
2025-06-02 08:33:24.996011924 +0000 UTC       |                    get_sint
2025-06-02 08:33:24.996050077 +0000 UTC /home/build/include/mysqlx/devapi/document.h: In member function 'T mysqlx::abi2::r0::Value::get() const [with T = long unsigned int]':
2025-06-02 08:33:24.996074985 +0000 UTC /home/build/include/mysqlx/devapi/document.h:811:12: error: 'get_uint' was not declared in this scope; did you mean 'get_sint'?
2025-06-02 08:33:24.996183435 +0000 UTC   811 |     return get_uint();
2025-06-02 08:33:24.996252245 +0000 UTC       |            ^~~~~~~~
2025-06-02 08:33:24.996274307 +0000 UTC       |            get_sint
2025-06-02 08:33:24.996319037 +0000 UTC /home/build/include/mysqlx/devapi/../common/value.h: In instantiation of 'mysqlx::abi2::r0::common::Value::Value(T) [with T = long unsigned int; typename std::enable_if<std::is_unsigned<_Tp>::value>::type* <anonymous> = 0]':
2025-06-02 08:33:24.996347354 +0000 UTC /home/build/include/mysqlx/devapi/document.h:741:22:   required from here
2025-06-02 08:33:24.996365822 +0000 UTC   741 |   : common::Value(val)
2025-06-02 08:33:24.996406862 +0000 UTC       |                      ^
2025-06-02 08:33:24.996434465 +0000 UTC /home/build/include/mysqlx/devapi/../common/value.h:168:21: error: expected primary-expression
2025-06-02 08:33:24.996457645 +0000 UTC   168 |     : Value(uint64_t(val))
2025-06-02 08:33:24.996495512 +0000 UTC       |             ~~~~~~~~^~~~~
2025-06-02 08:33:42.73326969 +0000 UTC ninja: subcommands failed
2025-06-02 08:33:42.73370573 +0000 UTC 
2025-06-02 08:33:45.871071046 +0000 UTC failed to build package: unable to run package mysql-connector-cpp pipeline: unable to run pipeline: unable to run pipeline: exit status 1
2025-06-02 08:33:45.979897486 +0000 UTC failed to build: command melange failed: exit status 1
2025-06-02 08:33:45.980126194 +0000 UTC build failed: command melange failed: exit status 1
```

--- a/mysql-connector-cpp.yaml
+++ b/mysql-connector-cpp.yaml
@@ -1,7 +1,7 @@
 package:
   name: mysql-connector-cpp
   version: "9.3.0"
-  epoch: 0
+  epoch: 1
   description: MySQL Connector/C++ is a MySQL database connector for C++
   copyright:
     - license: GPL-3.0-or-later
@@ -17,6 +17,7 @@ environment:
       - busybox
       - ca-certificates-bundle
       - cmake
+      - gcc-14-default
       - lz4-dev
       - openssl-dev
       - protobuf-dev
@@ -80,7 +81,7 @@ test:
     contents:
       packages:
         - mysql-connector-cpp-dev
-        - gcc
+        - gcc<15
         - glibc-dev
   pipeline:
     - uses: test/tw/ldd-check

